tpm2tools/seal_test.go

```diff
@@ -15,39 +15,60 @@ func TestSeal(t *testing.T) {
 	rwc := internal.GetTPM(t)
 	defer CheckedClose(t, rwc)
 
-	key, err := StorageRootKeyRSA(rwc)
-	if err != nil {
-		t.Fatalf("can't create srk from template: %v", err)
-	}
-	defer key.Close()
-
-	secret := []byte("test")
-	pcrList := []int{7, 23}
-	pcrToExtend := tpmutil.Handle(23)
-
-	sealed, err := key.Seal(pcrList, secret)
-	if err != nil {
-		t.Fatalf("failed to seal: %v", err)
+	type SRKey struct {
```

Comment:
So these table driven subtests are a little messy, one way to improve them would be use annonymous structs, and just pass the function that creates the key to the test. Specifically

```go
tests := []struct {
	name   string
	getSRK func(io.ReadWriter) (*Key, error)
}{
        {"RSA", StorageRootKeyRSA},
        {"ECC", StorageRootKeyECC},
}

for _, test := range tests {
        t.Run(test.name, func(t *testing.T) {
                srk, err := test.getSRK(rwc)
                // Check errors and use keys normally
        }
}
```

Reply:
Done